src/NServiceBus.Core/Notifications/NotificationExtensions.cs

```diff
@@ -0,0 +1,68 @@
+namespace NServiceBus
+{
+    using NServiceBus.Settings;
+
+    /// <summary>
+    /// Adds extensions methods to get notified of bus events.
+    /// </summary>
+    public static class NotificationExtensions
+    {
+
+        /// <summary>
+        /// Set a delegate that will be called when a message is moved to the error queue.
+        /// </summary>
+        public static void NotifyOnFailedMessage(this BusConfiguration busConfiguration, FailedMessageAction action)
+        {
+            Guard.AgainstNull(nameof(action), action);
+            busConfiguration.Settings.Set<FailedMessageAction>(action);
```

Comment:
"blocks us from using these notifications internally" why we own everything. we can do internal notifications any way we want